lib/Server.js

```diff
@@ -66,20 +66,29 @@ class Server {
 
     this.log = _log || createLogger(options);
 
-    // set serverMode default
-    if (this.options.serverMode === undefined) {
-      this.options.serverMode = 'sockjs';
-    } else {
-      this.log.warn(
-        'serverMode is an experimental option, meaning its usage could potentially change without warning'
-      );
-    }
-    // set clientMode default
-    if (this.options.clientMode === undefined) {
-      this.options.clientMode = 'sockjs';
+    if (this.options.transportMode === undefined) {
+      this.options.transportMode = {
+        server: 'sockjs',
+        client: 'sockjs',
+      };
     } else {
+      switch (typeof this.options.transportMode) {
+        case 'string':
+          this.options.transportMode = {
+            server: this.options.transportMode,
+            client: this.options.transportMode,
+          };
+          break;
+        // if not a string, it is an object
+        default:
+          this.options.transportMode.server =
+            this.options.transportMode.server || 'sockjs';
+          this.options.transportMode.client =
+            this.options.transportMode.client || 'sockjs';
+      }
+
       this.log.warn(
-        'clientMode is an experimental option, meaning its usage could potentially change without warning'
+        'transportMode is an experimental option, meaning its usage could potentially change without warning'
```

Comment:
Maybe we can move this in util, like `getTransportMode(options: object): object`